lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -2024,6 +2029,68 @@ bool MayaUsdProxyShapeBase::closestPoint(
 
 bool MayaUsdProxyShapeBase::canMakeLive() const { return (bool)_sharedClosestPointDelegate; }
 
+void _proxyShapeAncestorDirty(MObject& node, void* clientData)
+{
+    auto proxyShape = static_cast<MayaUsdProxyShapeBase*>(clientData);
+    if (proxyShape) {
+        proxyShape->onAncestorDirty();
+    }
+}
+
+void MayaUsdProxyShapeBase::clearAncestorCallbacks()
+{
+    for (auto cb : _ancestorCallbacks) {
+        MMessage::removeCallback(cb);
+    }
+
+    _ancestorCallbacks.clear();
+}
+
+void MayaUsdProxyShapeBase::updateAncestorCallbacks()
+{
+    clearAncestorCallbacks();
+
+    // Add our own callback
+    MObject obj = thisMObject();
+    _ancestorCallbacks.push_back(
+        MNodeMessage::addNodeDirtyCallback(obj, _proxyShapeAncestorDirty, this));
+
+    // Remember the path for which we are accumulating the callbacks
+    MDagPath ancestorPath;
+    MDagPath::getAPathTo(obj, ancestorPath);
+    _ancestorCallbacksPath = ancestorPath.fullPathName();
```

Comment:
What will happen if the proxy shape node is renamed? Or any of its ancestors are renamed?

Reply:
I just tested: renaming proxy shape node or any of its DAG ancestors works fine, as in this case we call updateAncestorCallbacks on the next dirty callback. However, after renaming a USD ancestor of a pulled node we loose the connection inside ProxyAcessors and this is a general problem that exists for all proxy accessors, including relaying the 3d transform. I think we may need to create a Jira ticket for that one.